### PR TITLE
[4.0] Convert inline to file and remove jQuery 

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue # .

### Summary of Changes
- moves inline script to a file
- removes the jQuery dependency
- Replaces `alert()` with `confirm()`
- Changes the logic: on confirm yes apply the same filter to all the child user groups

### Testing Instructions

Goto `administrator/index.php?option=com_config` and select the tab `Text Filters`
Apply `No Filtering` to `Author`
If you respond with yes to the popup then all the descending groups will also have a `No Filtering` value
if not then the rule will apply to the Author

### Documentation Changes Required

@brianteeman could you take a look at the string `COM_CONFIG_TEXT_FILTERS_NOTE`

Another observation here: the PHP should correctly calculate the selected filter depending on the parent if the selected value is no filtering (not what happens right now, which is completely confusing)

### CODE REVIEW

File: administrator/components/com_config/src/Field/FiltersField.php [hunk 1]
```diff
@@ -43,47 +43,11 @@ class FiltersField extends FormField
 	 */
 	protected function getInput()
 	{
-		// Load Framework
-		HTMLHelper::_('jquery.framework');
```
Comment (Quy) [inline administrator/components/com_config/src/Field/FiltersField.php:47]: Please remove:

`use Joomla\CMS\HTML\HTMLHelper;`

Comment (dgrammatiko) [reply administrator/components/com_config/src/Field/FiltersField.php:47]: Done

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 3177492b8c17358a5235e1f65274b683c8144ed5<br /><br />For Test: Using Firefox "OK" is "Yes", "Cancel" is "No":

image
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34478">issues.joomla.org/tracker/joomla-cms/34478</a>.</sub>

File: build/media_source/com_config/js/config-filters.es6.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+/**
+ * @copyright   (C) 2018 Open Source Matters, Inc. <https://www.joomla.org>
```
Comment (brianteeman) [inline build/media_source/com_config/js/config-filters.es6.js]:
```suggestion
 * @copyright   (C) 2021 Open Source Matters, Inc. <https://www.joomla.org>
```

**Comment by wilsonge** [comment]
Sorry dude can you sort conflicts here?

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**